.github/workflows/pr.yml

```diff
@@ -11,6 +11,40 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  build-binaries:
+    name: Build binaries
+    runs-on: ubuntu-latest
+    env:
+      ATHENA_DIR: "${{ github.workspace }}/.athena"
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+      - name: Install Athena toolchain
+        run: |
+          curl -L https://install.athenavm.org | bash
+          $ATHENA_DIR/bin/athup --path $GITHUB_WORKSPACE
+          $ATHENA_DIR/bin/cargo-athena athena --version
+          echo "$ATHENA_DIR/bin" >> $GITHUB_PATH
+
+      - name: Install Athena CLI
+        run: |
+          cd cli
+          cargo install --force --locked --path .
+          cargo athena install-toolchain
+
+      - name: Build binaries
+        run: |
+          make -C tests
+          make -C examples
+
+      - name: Upload host-test binary
+        uses: actions/upload-artifact@v4
```

Comment:
Is it worth caching these?

Reply:
Yes, I think so. Installation of the toolchain and building all tests and examples takes significant time. I've also seen toolchain installation fail because of exhausted HTTP query limits in the past. Installing it fewer times should help avoid this.

Reply:
Btw, it still happens unfortunately: https://github.com/athenavm/athena/actions/runs/11440975992/job/31829075665?pr=159

Reply:
don't you love that friendly, clear, helpful error message? :)